android/src/main/java/com/mattermost/turbolog/TurboLog.kt

```diff
@@ -0,0 +1,126 @@
+    private var logsDirectory: String? = null
+    private var configureOptions: ConfigureOptions? = null
+
+    fun getLogFiles(): Array<out File> {
+      val directory = logsDirectory?.let { File(it) }
+      return directory?.listFiles { _, name -> name.endsWith(".log") } ?: arrayOf()
+    }
+
+    fun reconfigure() {
+      configureOptions?.let { configure(it) }
+    }
+
+    fun configure(options: ConfigureOptions) {
+      val dailyRolling = options.dailyRolling ?: false
+      val maximumFileSize = options.maximumFileSize ?: (1024 * 1024)
+      val maximumNumberOfFiles = options.maximumNumberOfFiles ?: 2
+      logsDirectory = options.logsDirectory ?: ""
+      val logPrefix = options.logPrefix ?: ""
+
+      val loggerContext = LoggerFactory.getILoggerFactory() as LoggerContext
+
+      val rollingFileAppender = RollingFileAppender<ILoggingEvent>()
+      rollingFileAppender.context = loggerContext
+      rollingFileAppender.file = "$logsDirectory/$logPrefix-latest.log"
+
+      if (dailyRolling) {
+        val rollingPolicy = SizeAndTimeBasedRollingPolicy<ILoggingEvent>()
+        rollingPolicy.context = loggerContext
+        rollingPolicy.fileNamePattern = "$logsDirectory/$logPrefix-%d{yyyy-MM-dd}.%i.log"
+        rollingPolicy.setMaxFileSize(FileSize(maximumFileSize.toLong()))
+        rollingPolicy.setTotalSizeCap(FileSize(maximumNumberOfFiles.toLong() * maximumFileSize))
+        rollingPolicy.maxHistory = maximumNumberOfFiles
+        rollingPolicy.setParent(rollingFileAppender)
+        rollingPolicy.start()
+        rollingFileAppender.setRollingPolicy(rollingPolicy)
+      } else if (maximumFileSize > 0) {
+        val rollingPolicy = FixedWindowRollingPolicy()
+        rollingPolicy.context = loggerContext
+        rollingPolicy.fileNamePattern = "$logsDirectory/$logPrefix-%i.log"
+        rollingPolicy.minIndex = 1
+        rollingPolicy.maxIndex = maximumNumberOfFiles
+        rollingPolicy.setParent(rollingFileAppender)
+        rollingPolicy.start()
+        rollingFileAppender.rollingPolicy = rollingPolicy
+
+        val triggeringPolicy = SizeBasedTriggeringPolicy<ILoggingEvent>()
+        triggeringPolicy.context = loggerContext
+        triggeringPolicy.maxFileSize = FileSize(maximumFileSize.toLong())
+        triggeringPolicy.start()
+        rollingFileAppender.setTriggeringPolicy(triggeringPolicy)
+      }
+
+      val encoder = PatternLayoutEncoder()
+      encoder.context = loggerContext
+      encoder.charset = StandardCharsets.UTF_8
+      encoder.pattern = "%d{yyyy/MM/dd HH:mm:ss.SSS} %-5level %msg%n"
+      encoder.start()
+
+      rollingFileAppender.encoder = encoder
+      rollingFileAppender.start()
+
+      val root = LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME) as ch.qos.logback.classic.Logger
+      root.level = Level.DEBUG
+      root.detachAndStopAllAppenders()
+      root.addAppender(rollingFileAppender)
+
+      configureOptions = options
+    }
+
+    fun d(tag: String, message: String) {
```

Comment:
about the tag, we are defining a tag here https://github.com/mattermost/react-native-turbo-log/pull/6/files#diff-e816c369d11bba66419528c43d56b8f3d891038f4c578449c5366fda22bb6781R20 maybe we should just use that one for all the tags instead of allowing customization? thoughts?

Reply:
I had mixed feelings with this one, but I agree. Let's make the interface simpler.